packages/monitor-v2/src/monitor-og/oSnapAutomation.ts

```diff
@@ -242,6 +242,29 @@ const filterPotentialProposals = (
   });
 };
 
+// Filters out all Snapshot proposals that cannot be proposed due to blocking on-chain proposals. This is done by
+// matching safe and proposed transactions.
+const filterUnblockedProposals = async (
+  potentialProposals: SnapshotProposalExpanded[],
+  onChainProposals: TransactionsProposedEvent[],
+  params: MonitoringParams
+): Promise<SnapshotProposalExpanded[]> => {
+  // Filter out all on-chain proposals that have been executed since they cannot block new proposals.
+  const unexecutedProposals = await filterUnexecutedProposals(onChainProposals, params);
```

Comment:
If `filterUnexecutedProposals` errors this would raise an unhandled promise rejection. If there's a potential for this function to fail or throw an error, this could be handled.